third_party/nvidia/backend/compiler.py

```diff
@@ -231,7 +231,6 @@ def make_ttir(mod, metadata, opt):
         passes.ttir.add_combine(pm)
         passes.ttir.add_reorder_broadcast(pm)
         passes.common.add_cse(pm)
-        passes.common.add_licm(pm)
```

Comment:
why don't we do licm early anymore? In general we want to do it early and licm can be a problem later once op are decomposed. Did you measure the impact. If the problem is that doing licm on loops that will be fused is bad maybe we should explicitly skip those in our own licm pass?

Reply:
I think we could add our own LICM pass, but I didn't observe any issues with moving it. Although I have heard that there have been other contexts in which LICM was harmful. The problem with having LICM too early is that it can end up hoisting things out of inner loops and massively increasing register pressure. Doing it after flattening avoids this problem but it shouldn't change the effectiveness of LICM because ops haven't been lowered yet.

I measured the impact and it didn't seem to cause any issues.